GEECS-PythonAPI/geecs_python_api/controls/devices/geecs_device.py

```diff
@@ -121,10 +121,20 @@ def init_resources(self):
                     if self.dev_tcp is None:
                         try:
                             self.dev_tcp = TcpSubscriber(owner=self)
+                            ###try to connect a TCP subscriber to see if device is on and accessible
+                            is_device_on = self.dev_tcp.connect()
                         except Exception:
```

Comment:
When using a try-except block, especially one that is used as "if this one command raises an error" , be specific about the Exception you're catching. You want all other Exceptions to fail loudly.